src/pipecat/utils/text/markdown_text_filter.py

```diff
@@ -82,3 +99,114 @@ def filter(self, text: str) -> str:
             return filtered_text
         else:
             return text
+
+    def handle_interruption(self):
+        self._interrupted = True
+        self._in_code_block = False
+        self._in_table = False
+
+    def reset_interruption(self):
+        self._interrupted = False
+
+    #
+    # Filter code
+    #
+
+    def remove_code_blocks(self, text: str) -> str:
+        """
+        Main method to remove code blocks from the input text.
+        Handles interruptions and delegates to specific methods based on the current state.
+        """
+        if self._interrupted:
+            self._in_code_block = False
+            return text
+
+        # Pattern to match three consecutive backticks (code block delimiter)
+        code_block_pattern = r"```"
+        match = re.search(code_block_pattern, text)
+
+        if self._in_code_block:
+            return self._handle_in_code_block(match, text)
+
+        return self._handle_not_in_code_block(match, text, code_block_pattern)
+
+    def _handle_in_code_block(self, match, text):
+        """
+        Handle text when we're currently inside a code block.
+        If we find the end of the block, return text after it. Otherwise, skip the content.
+        """
+        if match:
+            self._in_code_block = False
+            end_index = match.end()
+            return text[end_index:].strip()
+        return " "  # Skip content inside code block
```

Comment:
why space?

Reply:
Not needed any more since I'm returning text in a compliant way with each service.